elastic/kibana
# [Security Solution] Expandable flyout - Rule preview contents

## Summary

This PR is part 2 of adding a rule preview panel to the expandable flyout. PR (https://github.com/elastic/kibana/pull/161999) adds the preview skeleton, and this PR populates the actual content related to rule details:

Expandable flyout:
- Updated title to include `created by` and `updated by` timestamps, and rule switch button
- Added contents for about, define, schedule and actions (if any)
- Added a hook to fetch data for rule switch button - logic mimics rule details page (`~/security_solution/public/detection_engine/rule_details_ui/pages/rule_details/index.tsx`)

Rule & detections:
- Added `isPanelView` option allow rendering rule details in smaller font, so that it can fit in panel view
- Minor UI updates to gutter sizes and spacing to accommodate long text
- Extracted `createdBy` and `updatedBy` to `~/security_solution/public/detections/components/rules/rule_info` to be shared between rule details page and flyout

![image](https://github.com/elastic/kibana/assets/18648970/bbccbec6-f5f2-4ac5-8715-9caf357283ee)

**How to test**
- add `xpack.securitySolution.enableExperimental: ['securityFlyoutEnabled']` to the `kibana.dev.json` file
- go to the Alerts page, and click on the expand detail button on any row of the table
- click on Overview, About, view Rule Summary, the rule preview panel should pop up

### Checklist
